scripts/git-append-commit-trailer

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+#
+# Append a trailer with the commit hash to every commit message.
+# This can be useful if you rewrite the history later on and you want
+# to preserve the original commit hashes.
+#
+# Attention: Since the commit message is part of the commit hash
+# calculation, this script changes the commit hashes too.
+#
+# Usage:
+# git-append-commit-trailer [<options>]
+#
+# Options:
+#   -f, --force   git filter-branch refuses to start with an existing
+#                 temporary directory or when there are already refs
+#                 starting with refs/original/, unless forced.
+#                 See `man git-filter-branch`
+#
```

Comment:
@larsxschneider I had to think twice about how to use it.
Maybe the example below makes its usage more clear? Maybe that's what @stoe referred to by adding a "production example"?

```suggestion
#
# Example:
#   Remove README.md file from repository:
#
#   1. git-append-commit-trailer
#   2. git filter-branch --tree-filter 'rm README.md' HEAD
#
#   Result: The history has been rewritten twice and the Git commit hashes have been changed twice as well.
#           The commit messages contain a commit trailer with the original commit hash.
#
#
```